pyqldb/driver/qldb_driver.py

```diff
@@ -234,25 +235,40 @@ def execute_lambda(self, query_lambda, retry_config=None):
 
         :raises LambdaAbortedError: If the lambda function calls :py:class:`pyqldb.execution.executor.Executor.abort`.
         """
-
         retry_config = self._retry_config if retry_config is None else retry_config
-        context = _LambdaExecutionContext()
-        get_session_attempt = 0
+
+        start_new_session = False
+        session = None
+        retry_attempt = 1
```

Comment:
If you set this to 0 on start, you can simplify the logic for your exception block to simply always have retry_attempt += 1 as the first line and not have it in two areas.

Reply:
Updated